### PR TITLE
Update pandas_extensions.py

### PR DESCRIPTION
**Pull request by 149189**
--- PR TEMPLATE INSTRUCTIONS (1) ---

Looking to submit a Hamilton Dataflow to the sf-hamilton-contrib module? If so go the the `Preview` tab and select the appropriate sub-template:
* [sf-hamilton-contrib template](?expand=1&template=HAMILTON_CONTRIB_PR_TEMPLATE.md)

Else, if not, please remove this block of text.

--- PR TEMPLATE INSTRUCTIONS (2) ---

[Short description explaining the high-level reason for the pull request]

## Changes
Removed the extra import pandas as pd.
Removed the redefined code.
The class PandasTableReader(DataLoader) is now decorated with dataclass.
pd.read_table is added.
Large Dict are made small.
Used the metadata utility like the others are using to provide metadata on the file being loaded.

## Checklist

- [ ] PR has an informative and human-readable title (this will be pulled into the release notes)
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future TODOs are captured in comments
- [ ] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

**Comment by 149189** [comment]
@skrawcz Hey! Can you review my code I made all the changes that you told bro.

File: hamilton/plugins/pandas_extensions.py [hunk 1]
```diff
@@ -983,6 +984,137 @@ def load_data(self, type: Type) -> Tuple[DATAFRAME_TYPE, Dict[str, Any]]:
...
+if __name__ == "__main__":
+    filepath = "pyarrow.feather"
+
+    reader = PandasTableReader(filepath)
+
+    print(PandasTableReader.name())
+
+    data, metadata = reader.load_data(DataFrameDataLoader)
+    print("Data loaded:", data)
+    print("Metadata:", metadata)
```
Comment (skrawcz) [inline hamilton/plugins/pandas_extensions.py:1114]: why is this code here? please remove it.

File: hamilton/plugins/pandas_extensions.py [hunk 2]
```diff
@@ -983,6 +984,137 @@ def load_data(self, type: Type) -> Tuple[DATAFRAME_TYPE, Dict[str, Any]]:
...
+    storage_options: str = None
+
+    reader_name: ClassVar[str] = "feather"
+
+    @classmethod
+    def name(cls) -> str:
+        return cls.reader_name
+
+    def _get_loading_kwargs(self) -> Dict[str, Any]:
+        kwargs = {
```
Comment (skrawcz) [inline hamilton/plugins/pandas_extensions.py:1045]: this does not look like the other classes. It should look something [like this](https://github.com/DAGWorks-Inc/hamilton/blob/main/hamilton/plugins/pandas_extensions.py#L265-L301).

File: hamilton/plugins/pandas_extensions.py [hunk 3]
```diff
@@ -983,6 +984,137 @@ def load_data(self, type: Type) -> Tuple[DATAFRAME_TYPE, Dict[str, Any]]:
...
+    low_memory: bool = True
+    memory_map: bool = False
+    float_precision: str = None
+    storage_options: str = None
+
+    reader_name: ClassVar[str] = "feather"
+
+    @classmethod
+    def name(cls) -> str:
+        return cls.reader_name
```
Comment (skrawcz) [inline hamilton/plugins/pandas_extensions.py:1042]: This should just be:

```python
    @classmethod
    def name(cls) -> str:
        return "feather"
```

File: hamilton/plugins/pandas_extensions.py [hunk 4]
```diff
@@ -983,6 +984,137 @@ def load_data(self, type: Type) -> Tuple[DATAFRAME_TYPE, Dict[str, Any]]:
...
+
+    def load_data(self, data_type: Type) -> Tuple[DataFrameDataLoader, Dict[str, Any]]:
+        filepath = utils.get_file_metadata(self.path)
+        kwargs = self._get_loading_kwargs()
+        data = pd.read_table(self.io, **self._get_loading_kwargs()) # Use pd.read_table
+
+        # Get metadata about the file being loaded
+        metadata = utils.get_file_metadata(self.path)
+
+        return data, metadata
```
Comment (skrawcz) [inline hamilton/plugins/pandas_extensions.py:1103]: this code does not work. Please correct.

**Review by skrawcz — CHANGES_REQUESTED** [review]
there is no unit test here. please also investigate and correct the test failures.

**Comment by skrawcz** [comment]
Let me know if you're still working on this else I will close this PR and reassign the issue. Thanks.

**Comment by skrawcz** [comment]
closing because I haven't heard back.

**Pull request closed without merging**